### PR TITLE
core: cross model interactions fixes for multi-actions

### PR DESCRIPTION
**Pull request by spolu**
## Description

- Workaround for Anthropic when other tool uses are present in the conversation
- Sanitization of tool_call_id for Misral which requires 9 alphanumerical characters (but call id can come from somewhere else).

## Risk

Low

## Deploy Plan

- deploy `core`

### CODE REVIEW

File: core/src/providers/anthropic.rs [hunk 1]
```diff
@@ -271,6 +273,41 @@ impl TryFrom<&ChatMessage> for AnthropicChatMessage {
...
+pub struct AnthropicToolChoice {
+    r#type: AnthropicToolChoiceType,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    name: Option<String>,
+}
+
+impl FromStr for AnthropicToolChoice {
+    type Err = ParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
```
Comment (fontanierh) [inline core/src/providers/anthropic.rs:294]: we don't want to parse the any ?

**Review by fontanierh — APPROVED** [review]
Looks great, just a few suggestions

File: core/src/providers/anthropic.rs (hunk 1 above)
Comment (spolu) [reply core/src/providers/anthropic.rs:294]: we don't support it officially so for now no.

File: core/src/providers/anthropic.rs [hunk 2]
```diff
@@ -671,6 +734,30 @@ impl AnthropicLLM {
 
         if !tools.is_empty() {
             body["tools"] = json!(tools);
+            if tool_choice.is_some() {
+                body["tool_choice"] = json!(tool_choice);
+            }
+        } else {
```
[the remark this answers is not in the document]
Comment (spolu) [reply core/src/providers/anthropic.rs:741]: done

File: core/src/providers/anthropic.rs (hunk 1 above)
Comment (fontanierh) [reply core/src/providers/anthropic.rs:294]: > we don't support it officially so for now no.

Agree that we don't support, but anthropic.rs doesn't really know that right ?

Comment (spolu) [reply core/src/providers/anthropic.rs:294]: fair

**Pull request merged**